lib/workers/repository/update/pr/code-owners.spec.ts

```diff
@@ -22,13 +23,183 @@ describe('workers/repository/update/pr/code-owners', () => {
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@jimmy', '@john']);
+      });
+
+      it('assigns changes in package C to @dan (c), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/c/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@dan', '@john']);
+      });
+
+      it('assigns changes in package D to @maria (d), @jimmy (d), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/d/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@maria', '@jimmy', '@john']);
+      });
+
+      it('assigns changes in package A and B to @maria (a), @jimmy (b), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/a/package.json',
+          'packages/b/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@maria', '@jimmy', '@john']);
+      });
+
+      it('assigns changes in package A, B and C to @john, @maria (a), @jimmy (b), @dan (c), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/a/package.json',
+          'packages/b/package.json',
+          'packages/c/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@maria', '@jimmy', '@dan', '@john']);
+      });
+
+      it('assigns changes in package C and D to @dan (c), @maria (d), @jimmy (e), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/c/package.json',
+          'packages/d/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@dan', '@maria', '@jimmy', '@john']);
+      });
+
+      it('assigns changes in package D and E to @jimmy (d, e), @maria (d), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/d/package.json',
+          'packages/e/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@jimmy', '@maria', '@john']);
+      });
+    });
+
+    it('does not require all files to match a single rule #12611', async () => {
```

Comment:
```suggestion
    it('does not require all files to match a single rule, regression test for #12611', async () => {
```

This looks like a regression test, might be good to mention this in the test description?